packages/core/src/routes/admin-user.ts

```diff
@@ -194,31 +199,59 @@ export default function adminUserRoutes<T extends AuthedRouter>(router: T) {
         body,
       } = ctx.guard;
 
-      await findUserById(userId);
+      const user = await findUserById(userId);
       await checkIdentifierCollision(body, userId);
 
+      const { roleNames, ...userUpdates } = body;
+
       // Temp solution to validate the existence of input roleNames
-      if (body.roleNames?.length) {
-        const { roleNames } = body;
+      if (roleNames) {
```

Comment:
while implementing the final solution of this later, will it be better to return early or wrap in a func instead have these nested `if`s?

Reply:
LOG-5046